CONTRIBUTING.md

```diff
@@ -0,0 +1,34 @@
+### Getting Started & Building:
+1. Prerequisites:
+   * CMake version 3.17.0+ -> Ubuntu systems seem to have a problem getting the latest version, please build from sources or use another package such as snap. 
+     * [Ubuntu CMake update instructions](https://apt.kitware.com/)
+   * Ninja
+2. Checkout the repository:
+    ```sh
+    git clone https://github.com/codereport/An-Algorithm-Library.git
+    ```
+3. Build An Algorithm Library:    
+   * Run cmake:
+    ```sh
+    cd An-Algorithm-Library
+    mkdir build
+    cmake -G "Ninja Multi-Config" -B build
+    ```
+   * To build debug:
+    ```sh
+    ninja -C build 
+    ```
+   * To build release:
+    ```sh
+    ninja -C build -f build-Release.ninja
+    ```
```

Comment:
Do we actually build something here? The only thing that should be compiled are the tests itself, which should be done in the test command.

Reply:
Oh, I see what you mean I can take that out. Though if you use the verbose output it requires the test to be built first. So I just moved the build command infront of that.

https://github.com/codereport/An-Algorithm-Library/pull/7/commits/7a275d579f579332c6c797b00611e1a742a7aafd